### ISSUE TITLE
Portable copy

### ISSUE THREAD

**Issue by cyberdyne-sys** [joncampbell123/dosbox-x#2154]
**Please describe your question.**
Sorry for the probably novice question, but I'm wondering, whether a portable setup means, is there any kind of activities of the emulator, outside of its own folder, or not at all? I mean system extensions, temporary files anywhere host-wide, registry entries and such? I need a clear developer confirmation. Thanks in advance.

**Comment by Wengier** [comment]
I believe you use Windows, and for Windows there are both Windows installer version and portable packages, as listed in the INSTALL page:

https://github.com/joncampbell123/dosbox-x/blob/master/INSTALL.md#windows-packages-installer-or-portable

If you use the Windows installer, some system resources such as the registry entries may be modified. However, if you use the portable packages, there will be no activates outside of its own folder at all, unless you decided to do so by yourself. Hope this answers your question.

**Comment by inukaze** [comment]
Hi there, the Windows Package of DOSBox-X is already a portable version.

For the GNU/Linux users, the thing is just a bit little diferent, because for example for make it portable i need make some thing first like
1 - Compile by myself the latest version of DOSBox-X under a mother GNU/Linux Distribution without things like : systemd, avahi, pulseaudio.

But with libraries like : xkb, sdl1.2, sdl2, cap, pcap, asound, fluidsynth, timidity, portautio, celt, sndfile, jack, termcap, nl, gen-nl, harfbuzz, libpng, X11 and freetype. 

The unique thing i can't compile under slackware is the ffmpeg, because is too old, and well i don't know which is the minumen version of ffmpeg for compile dosbox-x with that support.

2 - After compile the versions of SDL1 and SDL2, i must determine which libraries need the binary executable make it by myself, in another live distros, like Lubuntu, GoboLinux, ArtixLinux, etc . . . just for find the libraries the tool "ldd" don't tell me explicit and i ever put that libraries under a sub-folder called ".libs/32" for the 32 Bits binary executable and ".libs/64" for the 64 Bits binary executable. 

3 - After determine the libraries and test works fine the dosbox-x, i ever create a custom script can detect the environment arquitecture, set variables environement, launch the dosbox-x version for the right architecture.

And by that way is a form to make a Portable Copy for GNU/Linux users. the Windows users, just need files & folder: 
dosbox-x.exe, dosbox-x.conf, FREECG98.BMP and glshaders folder.

Ever can copy to another folder, make double click on executable and use it.

**Comment by rderooy** [comment]
@inukaze For Linux users there is a Flatpak, which is portable. But flatpaks do require that you have PulseAudio (or PipeWire) on the host for working sound.

**Comment by inukaze** [comment]
> @inukaze For Linux users there is a Flatpak, which is portable. But flatpaks do require that you have PulseAudio (or PipeWire) on the host for working sound.

Some distros don't use PulseAudio, i know is the optional standar, but all distros had ALSA because is part of kernel. 

The unique thing i don't understand in another proyects on GitHub, is the need to remove ALSA support but just had pulseaudio. 

i know PulseAudio can be the standard, but is a thing optional becuase is not part of kernel to make the audio works. For final users, its more like a modules for enable/disable access to some audio components like volume control, microphone, line-in, output. Came with the most popular GNU/Linux distros.

and the another thing, never a pack from FlatPak or Snap had work for me, the AppImage are most functionally and portables, like the FlatPak or Snap version. But i never can make my own AppImage of the DOSBox-X i had compile, on some point i ever fail.

**Comment by rderooy** [comment]
@inukaze we use the SDL2 provided by the flatpak base, and that one only has PulseAudio support enabled for passing audio through to the host. So I suggest you raise an issue with upstream flatpak to request that they support ALSA.

It is possible to run DOSBox-X flatpak on a system without PulseAudio by using the SDL dummy sound option.

**Comment by cyberdyne-sys** [comment]
@Wengier You're right, my OS is Windows. I just had to be absolutely sure, that it keeps the system fully untouched and intact and thanks to you, I got my answer.

Thanks to everyone for the precious comments as well.
